src/changelog/linker/mapper/leadingv.go

```diff
@@ -0,0 +1,82 @@
+package mapper
+
+import (
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Masterminds/semver"
+	"github.com/newrelic/release-toolkit/src/changelog"
+	"github.com/newrelic/release-toolkit/src/changelog/linker"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	checkTimeoutSeconds = 1
+)
+
+// LeadingVCheck wraps another mapper and performs a check over the result of Map. If the check is not
+// successful, it tries to generate it prepending a leading v to the version.
+type LeadingVCheck struct {
+	mapper    linker.Mapper
+	checkLink func(link string) bool
+}
+
+// WithLeadingVCheck returns a LeadingVCheck with the provided underlying mapper and a check function which
+// performs a request to the url corresponding to the link and check its status code.
+func WithLeadingVCheck(mapper linker.Mapper) *LeadingVCheck {
+	return &LeadingVCheck{mapper: mapper, checkLink: checkLinkResponse}
+}
```

Comment:
NIT
```suggestion
// NewWithLeadingVCheck returns a LeadingVCheck with the provided underlying mapper and a check function which
// performs a request to the url corresponding to the link and check its status code.
func NewWithLeadingVCheck(mapper linker.Mapper) *LeadingVCheck {
	return &LeadingVCheck{mapper: mapper, checkLink: checkLinkResponse}
}
```

Reply:
As far as I know, it is idiomatic to use `New` when you get a struct ready to be used in the same way constructors exist in OOP languages.

To have more than one constructor in Go we should use `NewWithWhatever`. Mapper here only has one constructor but if you want to be explicit I would prefer the `NewWithLeadingVCheck` in this case.

I am used to name functions that start with `With` for options using the options pattern: https://golang.cafe/blog/golang-functional-options-pattern.html

Reply:
Fair enough! 😃

Reply:
Updated here: [1315f5a](https://github.com/newrelic/release-toolkit/pull/134/commits/1315f5a7539fd21d8b4795257c35bc609d2bd1d5)